Makefile

```diff
@@ -75,10 +76,24 @@ local: ## - Build local binary for local environment (bin/fleet-server)
 	go build $(if $(DEV),-tags="dev",) -gcflags="${GCFLAGS}" -ldflags="${LDFLAGS}" -o ./bin/fleet-server .
 	@printf "${CMD_COLOR_ON} Binaries in ./bin/\n${CMD_COLOR_OFF}"
 
+
+.PHONY: cover-e2e-binaries
+cover-e2e-binaries: ## - Build binaries for the test-e2e target with the go 1.20+ cover flag
+	SNAPSHOT=true $(MAKE) cover-$(shell go env GOOS)/$(shell go env GOARCH) cover-linux/$(shell go env GOARCH)
+
+.PHONY: $(COVER_TARGETS)
+$(COVER_TARGETS): cover-%: ## - Build a binary with the -cover flag for integration testing
+	@mkdir -p build/cover
+	$(eval $@_OS := $(firstword $(subst /, ,$(lastword $(subst cover-, ,$@)))))
+	$(eval $@_GO_ARCH := $(lastword $(subst /, ,$(lastword $(subst cover-, ,$@)))))
+	$(eval $@_ARCH := $(TARGET_ARCH_$($@_GO_ARCH)))
+	$(eval $@_BUILDMODE:= $(BUILDMODE_$($@_OS)_$($@_GO_ARCH)))
+	GOOS=$($@_OS) GOARCH=$($@_GO_ARCH) go build $(if $(DEV),-tags="dev",) -cover -coverpkg=./... -gcflags="${GCFLAGS}" -ldflags="${LDFLAGS}" $($@_BUILDMODE) -o build/cover/fleet-server-$(VERSION)-$($@_OS)-$($@_ARCH)/fleet-server$(if $(filter windows,$($@_OS)),.exe,) .
```

Comment:
I think these binaries should be built using the docker `BUILDER_IMAGE`. Otherwise when these binaries are copied to the e2e testing images they are going to produce issues with glibc like the ones I am seeing:
```
/usr/share/elastic-agent/data/elastic-agent-abc162/components/fleet-server: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /usr/share/elastic-agent/data/elastic-agent-abc162/components/fleet-server)
```

Reply:
Done, I had to alter the `cover-e2e-binaries` target to make all binaries in order to also produce a local (mac/windows) one if it's not ran on a linux system.
I also adjusted the dockerfile itself to make this type of build easier